### ISSUE TITLE
Questions on store

### ISSUE THREAD

**Issue by blake41** [near-apps/nft-onboarding#1]
Hey Matt, this is great!  Very useful jumping off point. A few questions:
Is the store supposed to be accessible in any component, or is the idea that your top level component has access to it and you'd explicitly pass it down to any child component that needed it?
This seems pretty similar to using redux, any reason to use this model rather than redux?

**Comment by mattlockyer** [comment]
Hi @blake41 I don't use redux because it's adding a lot of extra complexity for what is essentially a boilerplate.

I use the React 17 useContext, useReducer hooks with some additional code I added to manage app state in a way that I felt was ergonomic.

Foundation of the nearbp (which this example is based on) is from here:
https://github.com/mattlockyer/rpbp/
nearbp
https://github.com/near-apps/nearbp


This has nothing to do with NEAR though, and your choice how you manage your state or where you expose the contract instance, near-api-js etc... is up to you.

**Comment by blake41** [comment]
Makes sense!  I like the idea of you guys allowing everyone to decide how to manage state.  I also would love to see a canonical implementation NEAR recommends (maybe bundled with the create-near-app) that allows access to the app state containing a reference to NEAR.  [This seems like a pretty well thought out way to do it](https://github.com/mehtaphysical/near-react-hooks) and allows a lot of freedom and not much boilerplate like redux.
